### PR TITLE
Build Separation Stage2

### PR DESCRIPTION
**Pull request by brthor**
This change moves packaging/publishing of the host and shared framework to the host build, except for the finalization step.

This is the next step to breaking apart the repos.

cc @Sridhar-MS @eerhardt @schellap

### CODE REVIEW

File: build_projects/dotnet-cli-build/MsiTargets.cs [hunk 1]
```diff
@@ -202,61 +144,10 @@ public static BuildTargetResult GenerateCliSdkBundle(BuildTargetContext c)
...
-            return c.Success();
-        }
-
         private static string GetEngineName(string bundle)
         {
             var engine = $"{Path.GetFileNameWithoutExtension(bundle)}-{ENGINE}";
             return Path.Combine(Path.GetDirectoryName(bundle), engine);
         }
-
-        private static void ExtractEngineFromBundleHelper(string bundle, string engine)
```
Comment (brthor) [inline build_projects/dotnet-cli-build/MsiTargets.cs]: @Sridhar-MS this looked to be dead code, can you confirm?

File: build_projects/dotnet-host-build/MsiTargets.cs [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+        public static BuildTargetResult GenerateSharedFxBundle(BuildTargetContext c)
+        {
+            var sharedFrameworkNuGetName = Monikers.SharedFrameworkName;
+            var sharedFrameworkNuGetVersion = c.BuildContext.Get<string>("SharedFrameworkNugetVersion");
+            var upgradeCode = Utils.GenerateGuidFromName(SharedFrameworkBundle).ToString().ToUpper();
+            var sharedFxBrandName = $"'{Monikers.SharedFxBrandName}'";
+
+            Cmd("powershell", "-NoProfile", "-NoLogo",
+                Path.Combine(Dirs.RepoRoot, "packaging", "windows", "sharedframework", "generatebundle.ps1"),
+                SharedFrameworkMsi, SharedHostMsi, SharedFrameworkBundle, WixRoot, sharedFxBrandName, MsiVersion, CliDisplayVersion, sharedFrameworkNuGetName, sharedFrameworkNuGetVersion, upgradeCode, Arch)
```
Comment (brthor) [inline build_projects/dotnet-host-build/MsiTargets.cs]: @Sridhar-MS Can you help me understand why this requires the CliDisplayVersion?

File: build_projects/Microsoft.DotNet.Cli.Build.Framework/project.json [hunk 3]
```diff
@@ -1,5 +1,6 @@
 {
   "version": "1.0.0-*",
+
```
Comment (eerhardt) [inline build_projects/Microsoft.DotNet.Cli.Build.Framework/project.json]: These changes can probably be reverted.  The next time update-dependencies runs it will remove the blank lines anyway.

File: build_projects/dotnet-cli-build/CompileTargets.cs [hunk 4]
```diff
@@ -15,7 +15,6 @@ namespace Microsoft.DotNet.Cli.Build
 {
     public class CompileTargets
     {
-        public static readonly string CoreCLRVersion = "1.0.2-rc2-24027";
```
Comment (eerhardt) [inline build_projects/dotnet-cli-build/CompileTargets.cs]: You will need to update the update-dependencies script.

File: build_projects/dotnet-cli-build/CompileTargets.cs [hunk 5]
```diff
@@ -219,143 +216,7 @@ private static void CopySharedHost(string outputDir)
...
-            sharedFrameworkProject["runtimes"][rid] = new JObject();
-
-            string projectJsonPath = Path.Combine(sharedFrameworkProjectPath, "project.json");
-            JsonUtils.WriteProject(sharedFrameworkProject, projectJsonPath);
-
-            Rm(templateFile);
-
-            return sharedFrameworkProjectPath;
-        }
+
```
Comment (eerhardt) [inline build_projects/dotnet-cli-build/CompileTargets.cs]: nit: double blank line

File: build_projects/dotnet-cli-build/CompileTargets.cs [hunk 6]
```diff
@@ -418,11 +279,23 @@ private static BuildTargetResult CompileCliSdk(BuildTargetContext c, DotNetCli d
             }
 
             // cleanup compilers project output we don't need
-            DeleteMainPublishOutput(outputDir, "compilers");
+            PublishMutationUtilties.CleanPublishOutput(outputDir, "compilers");
             File.Delete(compilersDeps);
-            File.Delete(compilersRuntimeConfig);
```
Comment (eerhardt) [inline build_projects/dotnet-cli-build/CompileTargets.cs]: ?? Do you really want this line removed?

File: build_projects/dotnet-cli-build/dotnet-cli-build.xproj [hunk 7]
```diff
@@ -1,14 +1,14 @@
...
-    <VisualStudioVersion Condition="'$(VisualStudioVersion)' == ''">14.0</VisualStudioVersion>
+    <VisualStudioVersion Condition="'$(VisualStudioVersion)' == ''">14.0.25123</VisualStudioVersion>
     <VSToolsPath Condition="'$(VSToolsPath)' == ''">$(MSBuildExtensionsPath32)\Microsoft\VisualStudio\v$(VisualStudioVersion)</VSToolsPath>
   </PropertyGroup>
   <Import Project="$(VSToolsPath)\DNX\Microsoft.DNX.Props" Condition="'$(VSToolsPath)' != ''" />
   <PropertyGroup Label="Globals">
     <ProjectGuid>d7b9695d-23eb-4ea8-b8ab-707a0092e1d5</ProjectGuid>
     <RootNamespace>Microsoft.DotNet.Cli.Build</RootNamespace>
-    <BaseIntermediateOutputPath Condition="'$(BaseIntermediateOutputPath)'=='' ">..\..\artifacts\obj\$(MSBuildProjectName)</BaseIntermediateOutputPath>
+    <BaseIntermediateOutputPath Condition="'$(BaseIntermediateOutputPath)'=='' ">..\..\artifacts\obj$(MSBuildProjectName)</BaseIntermediateOutputPath>
```
Comment (eerhardt) [inline build_projects/dotnet-cli-build/dotnet-cli-build.xproj]: Was this change intended?

File: build_projects/dotnet-cli-build/project.json [hunk 8]
```diff
@@ -13,15 +13,27 @@
     "System.Xml.XmlSerializer": "4.0.11-rc2-24027",
     "WindowsAzure.Storage": "6.2.2-preview",
 
-    "Microsoft.DotNet.Cli.Build.Framework": {"target":"project"},
+    "Microsoft.DotNet.Cli.Build.Framework": "1.0.0-*",
```
Comment (eerhardt) [inline build_projects/dotnet-cli-build/project.json]: I think this should stay {target: project}.  From what I've heard that is the correct way to P2P reference.

File: build_projects/dotnet-cli-build/project.json [hunk 9]
```diff
@@ -13,15 +13,27 @@
...
     "shared-build-targets-utils": {"target": "project"}
   },
   "frameworks": {
     "netcoreapp1.0": {
       "imports": [
         "dnxcore50",
-        "portable-net45+win8"
+        "portable-net45+win8",
+        "netstandard1.3"
```
Comment (eerhardt) [inline build_projects/dotnet-cli-build/project.json]: I don't think this is necessary.  netcoreapp references netstandard1.3 automatically.

File: build_projects/dotnet-host-build/CompileTargets.cs [hunk 10]
```diff
@@ -38,13 +40,12 @@ public class CompileTargets
         };
```
Comment (eerhardt) [inline build_projects/dotnet-host-build/CompileTargets.cs]: nit: lots of blank lines

File: build_projects/dotnet-cli-build/CompileTargets.cs [hunk 11]
```diff
@@ -418,11 +279,23 @@ private static BuildTargetResult CompileCliSdk(BuildTargetContext c, DotNetCli d
...
+            
+            // Publish SharedFx
+            var sharedFrameworkNugetVersion = c.BuildContext.Get<string>("SharedFrameworkNugetVersion");
+            var commitHash = c.BuildContext.Get<string>("CommitHash");
+
+            var sharedFrameworkPublisher = new SharedFrameworkPublisher(
```
Comment (eerhardt) [inline build_projects/dotnet-cli-build/CompileTargets.cs]: Why are we publishing the shared fx twice?  Once here and once in dotnet-host-build.

File: build_projects/dotnet-host-build/MsiTargets.cs (hunk 2 above)
Comment (eerhardt) [inline build_projects/dotnet-host-build/MsiTargets.cs]: That has to be a bug.  The shared fx should have no knowledge of CLI anything.

File: build_projects/dotnet-host-build/PublishTargets.cs [hunk 12]
```diff
@@ -0,0 +1,243 @@
...
+                string source = blob.Replace("/dotnet/", "");
+                string targetName = Path.GetFileName(blob)
+                                        .Replace(SharedFrameworkNugetVersion, "latest")
+                                        .Replace(SharedHostNugetVersion, "latest");
+                string target = $"{destinationFolder}{targetName}";
+                AzurePublisherTool.CopyBlob(source, target);
+            }
+        }
+
+        private static bool CheckIfAllBuildsHavePublished()
```
Comment (eerhardt) [inline build_projects/dotnet-host-build/PublishTargets.cs]: Is this dead code?

File: build_projects/dotnet-host-build/PublishTargets.cs [hunk 13]
```diff
@@ -0,0 +1,243 @@
...
+        [Target(nameof(PrepareTargets.Init),
+        nameof(PublishTargets.InitPublish),
+        nameof(PublishTargets.PublishArtifacts))]
+        [Environment("PUBLISH_TO_AZURE_BLOB", "1", "true")] // This is set by CI systems
+        public static BuildTargetResult Publish(BuildTargetContext c)
+        {
+            return c.Success();
+        }
+
+        private static void CopyBlobs(string sourceFolder, string destinationFolder)
```
Comment (eerhardt) [inline build_projects/dotnet-host-build/PublishTargets.cs]: Dead?

File: build_projects/dotnet-host-build/dotnet-host-build.xproj [hunk 14]
```diff
@@ -10,6 +10,7 @@
     <RootNamespace>Microsoft.DotNet.Cli.Build</RootNamespace>
     <BaseIntermediateOutputPath Condition="'$(BaseIntermediateOutputPath)'=='' ">..\..\artifacts\obj\$(MSBuildProjectName)</BaseIntermediateOutputPath>
     <OutputPath Condition="'$(OutputPath)'=='' ">..\..\artifacts\bin</OutputPath>
+
```
Comment (eerhardt) [inline build_projects/dotnet-host-build/dotnet-host-build.xproj]: This file can be reverted.

File: build_projects/dotnet-host-build/project.json [hunk 15]
```diff
@@ -15,7 +15,6 @@
 
     "Microsoft.DotNet.Cli.Build.Framework": {"target":"project"},
     "shared-build-targets-utils": {"target": "project"}
-
```
Comment (eerhardt) [inline build_projects/dotnet-host-build/project.json]: revert

File: build_projects/shared-build-targets-utils/BuildCoreClrVersion.cs [hunk 16]
```diff
@@ -0,0 +1,12 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Microsoft.DotNet.Cli.Build
+{
+    public class BuildPackageVersions
+    {
+        public static readonly string CoreCLRVersion = "1.0.2-rc2-24027";
```
Comment (eerhardt) [inline build_projects/shared-build-targets-utils/BuildCoreClrVersion.cs]: Make sure to update update-dependencies to change this when we bring in a new CoreCLR version.

File: build_projects/shared-build-targets-utils/Utils/Crossgen.cs [hunk 17]
```diff
@@ -62,17 +62,15 @@ private string GetCrossgenPathForVersion()
...
         }
 
-        public void CrossgenDirectory(BuildTargetContext c, string pathToAssemblies)
+        public void CrossgenDirectory(string sharedFxPath, string pathToAssemblies)
         {
             // Check if we need to skip crossgen
             if (string.Equals(Environment.GetEnvironmentVariable("DISABLE_CROSSGEN"), "1"))
             {
-                c.Warn("Skipping crossgen for because DISABLE_CROSSGEN is set to 1");
+                Console.WriteLine("Skipping crossgen for because DISABLE_CROSSGEN is set to 1");
```
Comment (eerhardt) [inline build_projects/shared-build-targets-utils/Utils/Crossgen.cs]: Can you make sure this stays .Yellow() so it stands out?

File: build_projects/shared-build-targets-utils/project.json [hunk 18]
```diff
@@ -3,12 +3,17 @@
   "description": "Build scripts for dotnet-cli",
   "dependencies": {
     "NETStandard.Library": "1.5.0-rc2-24027",
+
```
Comment (eerhardt) [inline build_projects/shared-build-targets-utils/project.json]: This file can be reverted.

File: build_projects/shared-build-targets-utils/Utils/Crossgen.cs (hunk 17 above)
Comment (brthor) [inline build_projects/shared-build-targets-utils/Utils/Crossgen.cs]: Yes we can set console color. I don't think we should be passing in the build context here.

File: build_projects/shared-build-targets-utils/BuildCoreClrVersion.cs (hunk 16 above)
Comment (brthor) [inline build_projects/shared-build-targets-utils/BuildCoreClrVersion.cs]: Yes, I remembered :) 

That is coming in a future PR, I have been moving this thing around a lot, but it should stay here.

File: build_projects/dotnet-cli-build/MsiTargets.cs (hunk 1 above)
Comment (Sridhar-MS) [inline build_projects/dotnet-cli-build/MsiTargets.cs]: This is needed for signing purposes, Do not remove this

File: build_projects/dotnet-host-build/PublishTargets.cs (hunk 13 above)
Comment (brthor) [inline build_projects/dotnet-host-build/PublishTargets.cs]: More work to do here. WIP

File: build_projects/dotnet-host-build/PublishTargets.cs (hunk 12 above)
Comment (brthor) [inline build_projects/dotnet-host-build/PublishTargets.cs]: More work to do here. WIP

File: build_projects/dotnet-host-build/MsiTargets.cs (hunk 2 above)
Comment (brthor) [inline build_projects/dotnet-host-build/MsiTargets.cs]: I'm going to need to dig into these scripts

File: build_projects/dotnet-cli-build/CompileTargets.cs (hunk 6 above)
Comment (brthor) [inline build_projects/dotnet-cli-build/CompileTargets.cs]: Yes

File: build_projects/dotnet-cli-build/CompileTargets.cs (hunk 11 above)
Comment (brthor) [inline build_projects/dotnet-cli-build/CompileTargets.cs]: They are used for separate things, dotnet-host-build publishes it to a different directory to create zip packages, dotnet-cli-build publishes it to the stage directories to create appropriately constructed, usable layouts.

File: build_projects/dotnet-cli-build/CompileTargets.cs (hunk 5 above)
Comment (brthor) [inline build_projects/dotnet-cli-build/CompileTargets.cs]: will clean

File: build_projects/Microsoft.DotNet.Cli.Build.Framework/project.json (hunk 3 above)
Comment (brthor) [inline build_projects/Microsoft.DotNet.Cli.Build.Framework/project.json]: okay

File: build_projects/dotnet-cli-build/dotnet-cli-build.xproj (hunk 7 above)
Comment (brthor) [inline build_projects/dotnet-cli-build/dotnet-cli-build.xproj]: I'm going to revert all of these xprojs.. I was experimenting with a few things when I couldn't get things working.

File: build_projects/dotnet-cli-build/project.json (hunk 8 above)
Comment (brthor) [inline build_projects/dotnet-cli-build/project.json]: Same here, this is from experimentation when I couldn't get P2Ps working in VS. These weird changes will get reverted.

File: build_projects/dotnet-cli-build/project.json (hunk 9 above)
Comment (brthor) [inline build_projects/dotnet-cli-build/project.json]: Same here, this is from experimentation when I couldn't get P2Ps working in VS. These weird changes will get reverted.

File: build_projects/dotnet-host-build/PublishTargets.cs (hunk 13 above)
Comment (brthor) [inline build_projects/dotnet-host-build/PublishTargets.cs]: But yes this is going away.

File: build_projects/dotnet-host-build/PublishTargets.cs (hunk 12 above)
Comment (brthor) [inline build_projects/dotnet-host-build/PublishTargets.cs]: But yes this is going away.

File: build_projects/dotnet-host-build/dotnet-host-build.xproj (hunk 14 above)
Comment (brthor) [inline build_projects/dotnet-host-build/dotnet-host-build.xproj]: will do

File: build_projects/dotnet-host-build/project.json (hunk 15 above)
Comment (brthor) [inline build_projects/dotnet-host-build/project.json]: will do

File: build_projects/shared-build-targets-utils/project.json (hunk 18 above)
Comment (brthor) [inline build_projects/shared-build-targets-utils/project.json]: will do.

File: build_projects/dotnet-cli-build/MsiTargets.cs (hunk 1 above)
Comment (brthor) [inline build_projects/dotnet-cli-build/MsiTargets.cs]: 👍

File: build_projects/dotnet-cli-build/CompileTargets.cs (hunk 11 above)
Comment (eerhardt) [inline build_projects/dotnet-cli-build/CompileTargets.cs]: Shouldn't it be published once in `dotnet-host-build`, and then `dotnet-cli-build` take what was published during `dotnet-host-build` and use it?

I'm thinking when these are split repos:

- `host-build` publishes the shared fx
- `host-build` creates the .zip and publishes it to Azure
- `cli-build` pulls the .zip, and extracts it locally

Or were you thinking it would work differently?

Comment (brthor) [inline build_projects/dotnet-cli-build/CompileTargets.cs]: No I agree that's how it would work. 

I hadn't yet put in the logic to pull the .zip in the cli-build so publishing to each stage still made sense. I agree that it should pull this and just copy it into `stage1` and `stage2`. 

Since we've come to conclusions on the repo structure, it makes sense to make that change immediately no, I think.

File: build_projects/dotnet-cli-build/CompileTargets.cs (hunk 4 above)
Comment (brthor) [inline build_projects/dotnet-cli-build/CompileTargets.cs]: done.

File: build_projects/dotnet-host-build/CompileTargets.cs (hunk 10 above)
Comment (brthor) [inline build_projects/dotnet-host-build/CompileTargets.cs]: fixed.

File: build_projects/shared-build-targets-utils/BuildCoreClrVersion.cs (hunk 16 above)
Comment (brthor) [inline build_projects/shared-build-targets-utils/BuildCoreClrVersion.cs]: updated.

File: build_projects/shared-build-targets-utils/Utils/Crossgen.cs (hunk 17 above)
Comment (brthor) [inline build_projects/shared-build-targets-utils/Utils/Crossgen.cs]: fixed.

**Comment by brthor** [comment]
This is currently blocked by https://github.com/NuGet/Home/issues/2818

Working on finding a workaround..

**Comment by brthor** [comment]
Found a workaround

File: build_projects/dotnet-host-build/MsiTargets.cs (hunk 2 above)
Comment (brthor) [inline build_projects/dotnet-host-build/MsiTargets.cs]: Dug through this, `CLIDisplayVersion` get's passes as the version of the bundle, but it is a misnomer, it really represents the 4-part build number. This is just a rename.

File: NuGet.Config [hunk 19]
```diff
@@ -5,5 +5,6 @@
     <clear />
     <add key="cli-deps" value="https://dotnet.myget.org/F/cli-deps/api/v3/index.json" />
     <add key="api.nuget.org" value="https://api.nuget.org/v3/index.json" />
+    <add key="cli" value="https://dotnet.myget.org/F/dotnet-cli/api/v3/index.json" />
```
Comment (brthor) [inline NuGet.Config]: will remove this.

**Comment by brthor** [comment]
Please take another pass at reviews, waiting on builds but code is nearly complete.

File: build.cmd [hunk 20]
```diff
@@ -4,4 +4,7 @@ REM Copyright (c) .NET Foundation and contributors. All rights reserved.
 REM Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
 powershell -NoProfile -NoLogo -Command "%~dp0build_projects\dotnet-host-build\build.ps1 %*; exit $LastExitCode;"
+if %errorlevel% neq 0 exit /b %errorlevel%
```
Comment (eerhardt) [inline build.cmd]: Do this on *nix as well?

File: build_projects/dotnet-host-build/CompileTargets.cs [hunk 21]
```diff
@@ -37,14 +39,11 @@ public class CompileTargets
...
-
-        private static string DotnetHostBaseName => $"dotnet{Constants.ExeSuffix}";
-        private static string DotnetHostFxrBaseName => $"{Constants.DynamicLibPrefix}hostfxr{Constants.DynamicLibSuffix}";
-        private static string HostPolicyBaseName => $"{Constants.DynamicLibPrefix}hostpolicy{Constants.DynamicLibSuffix}";
-
         [Target(nameof(PrepareTargets.Init), 
             nameof(CompileCoreHost), 
-            nameof(PackagePkgProjects))]
+            nameof(PackagePkgProjects),
+            nameof(RestoreLockedCoreHost),
```
Comment (eerhardt) [inline build_projects/dotnet-host-build/CompileTargets.cs]: I thought you and Senthil said last time that RestoreLockedCoreHost was supposed to be in dotnet-cli-build?  [https://github.com/dotnet/cli/pull/3003#discussion_r62949648](https://github.com/dotnet/cli/pull/3003#discussion_r62949648)

File: build_projects/dotnet-cli-build/CompileTargets.cs [hunk 22]
```diff
@@ -418,71 +222,27 @@ private static BuildTargetResult CompileCliSdk(BuildTargetContext c, DotNetCli d
             }
 
             // cleanup compilers project output we don't need
-            DeleteMainPublishOutput(outputDir, "compilers");
+            PublishMutationUtilties.CleanPublishOutput(sdkOutputDirectory, "compilers");
             File.Delete(compilersDeps);
-            File.Delete(compilersRuntimeConfig);
```
Comment (eerhardt) [inline build_projects/dotnet-cli-build/CompileTargets.cs]: Should CleanPublishOutput also remove the .deps file so the above line can go away as well?

File: build_projects/dotnet-cli-build/DebTargets.cs [hunk 23]
```diff
@@ -12,9 +12,7 @@ namespace Microsoft.DotNet.Cli.Build
 {
     public class DebTargets
     {
-        [Target(nameof(GenerateSharedHostDeb),
```
Comment (eerhardt) [inline build_projects/dotnet-cli-build/DebTargets.cs]: Should these methods get removed from this file as well?

**Comment by eerhardt** [comment]
>             SharedFxComponentId = $"com.microsoft.dotnet.sharedframework.{sharedFrameworkNugetName}.{SharedFrameworkNugetVersion}.component.osx.x64";

Do we need these "SharedFx" variables?

---
Refers to: build_projects/dotnet-cli-build/PkgTargets.cs:35 in 2d8a4bb. [](commit_id = 2d8a4bb70d98454da9cbf3213a926c147c0a97a6, deletion_comment = False)

File: build_projects/dotnet-cli-build/project.json [hunk 24]
```diff
@@ -13,8 +13,8 @@
     "System.Xml.XmlSerializer": "4.0.11-rc2-24027",
     "WindowsAzure.Storage": "6.2.2-preview",
 
-    "Microsoft.DotNet.Cli.Build.Framework": { "target": "project" },
```
Comment (eerhardt) [inline build_projects/dotnet-cli-build/project.json]: I'd revert your changes to this whole file.

**Comment by eerhardt** [comment]
>         public static BuildTargetResult PublishCoreHostPackages(BuildTargetContext c)

Should this be removed?  It is in the host build.

---
Refers to: build_projects/dotnet-cli-build/PublishTargets.cs:221 in 2d8a4bb. [](commit_id = 2d8a4bb70d98454da9cbf3213a926c147c0a97a6, deletion_comment = False)

File: build_projects/dotnet-host-build/PrepareTargets.cs [hunk 25]
```diff
@@ -173,6 +231,19 @@ public static BuildTargetResult CheckCentOSCoreclrAndCoreFxDependencies(BuildTar
         }
 
         [Target]
+        public static BuildTargetResult RestorePackages(BuildTargetContext c)
+        {
+            var dotnet = DotNetCli.Stage0;
+
+            dotnet.Restore("--verbosity", "verbose", "--disable-parallel", "--infer-runtimes")
```
Comment (eerhardt) [inline build_projects/dotnet-host-build/PrepareTargets.cs]: do we need --infer-runtimes?  Can the runtimes be put in the PJ?

File: tools/independent/RuntimeGraphGenerator/Program.cs [hunk 26]
```diff
@@ -0,0 +1,105 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
```
Comment (eerhardt) [inline tools/independent/RuntimeGraphGenerator/Program.cs]: Are these all "dupes" of the existing RuntimeGraphGenerator?  Can these files be moved instead?

File: tools/RuntimeGraphGenerator/global.json [hunk 27]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (eerhardt) [inline tools/RuntimeGraphGenerator/global.json]: I assume this was the 'non-working' global.json.  Can this file be reverted?

File: build_projects/shared-build-targets-utils/project.json [hunk 28]
```diff
@@ -3,12 +3,17 @@
...
     "Microsoft.CSharp": "4.0.1-rc2-24027",
+
     "Microsoft.DotNet.Cli.Build.Framework": "1.0.0-*",
+
     "System.Dynamic.Runtime": "4.0.11-rc2-24027",
     "System.Reflection.Metadata": "1.3.0-rc2-24027",
     "System.Runtime.Serialization.Primitives": "4.1.1-rc2-24027",
     "System.Xml.XmlSerializer": "4.0.11-rc2-24027",
+
```
Comment (eerhardt) [inline build_projects/shared-build-targets-utils/project.json]: This file can still be reverted.

File: build_projects/shared-build-targets-utils/BuildPackageVersions.cs [hunk 29]
```diff
@@ -0,0 +1,12 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Microsoft.DotNet.Cli.Build
+{
+    public class BuildPackageVersions
```
Comment (eerhardt) [inline build_projects/shared-build-targets-utils/BuildPackageVersions.cs]: How about "DependencyVersions"?  I'm not sure I understand what "BuildPackage"s are.

File: build.cmd (hunk 20 above)
Comment (brthor) [inline build.cmd]: nix uses `set -e` which is more encompassing.

File: build_projects/dotnet-cli-build/DebTargets.cs (hunk 23 above)
Comment (brthor) [inline build_projects/dotnet-cli-build/DebTargets.cs]: will do. good catch.

File: build_projects/dotnet-cli-build/project.json (hunk 24 above)
Comment (brthor) [inline build_projects/dotnet-cli-build/project.json]: looks like a bad rebase.

File: build_projects/dotnet-host-build/CompileTargets.cs (hunk 21 above)
Comment (brthor) [inline build_projects/dotnet-host-build/CompileTargets.cs]: When we were publishing the shared framework in the cli build it made sense. Now the cli build uses the layout produced in the host build and eventually the zips.

File: build_projects/dotnet-host-build/PrepareTargets.cs (hunk 25 above)
Comment (brthor) [inline build_projects/dotnet-host-build/PrepareTargets.cs]: I can remove this.

File: build_projects/shared-build-targets-utils/BuildPackageVersions.cs (hunk 29 above)
Comment (brthor) [inline build_projects/shared-build-targets-utils/BuildPackageVersions.cs]: fair enough

File: build_projects/shared-build-targets-utils/project.json (hunk 28 above)
Comment (brthor) [inline build_projects/shared-build-targets-utils/project.json]: missed this one.

File: tools/independent/RuntimeGraphGenerator/Program.cs (hunk 26 above)
Comment (brthor) [inline tools/independent/RuntimeGraphGenerator/Program.cs]: These are the right files, I didn't commit the deletion of the existing one.

File: tools/RuntimeGraphGenerator/global.json (hunk 27 above)
Comment (brthor) [inline tools/RuntimeGraphGenerator/global.json]: This whole directory needs to be deleted.

File: build_projects/dotnet-cli-build/CompileTargets.cs (hunk 22 above)
Comment (brthor) [inline build_projects/dotnet-cli-build/CompileTargets.cs]: I thought about this, but the deps file is not needing to be deleted in the other place it is used. I suppose I could make it optional.

File: build_projects/dotnet-host-build/CompileTargets.cs (hunk 21 above)
Comment (brthor) [inline build_projects/dotnet-host-build/CompileTargets.cs]: this relates back to our discussion yesterday on the two-step process. This isolates the decision of which version of the host should be used to the sharedfx portion of the build.

File: tools/independent/RuntimeGraphGenerator/Program.cs (hunk 26 above)
Comment (brthor) [inline tools/independent/RuntimeGraphGenerator/Program.cs]: the reason they had to move was to add the global.json in the `independent` directory. I thought the name was fitting as these are the tools which leverage the global.json to be independent of project references. Admittedly it's not the best name though, so if you have a suggestion feel free.

Comment (brthor) [inline tools/independent/RuntimeGraphGenerator/Program.cs]: fixed.

File: tools/RuntimeGraphGenerator/global.json (hunk 27 above)
Comment (brthor) [inline tools/RuntimeGraphGenerator/global.json]: fixed.

File: build_projects/shared-build-targets-utils/project.json (hunk 18 above)
Comment (brthor) [inline build_projects/shared-build-targets-utils/project.json]: done.

File: build_projects/shared-build-targets-utils/BuildPackageVersions.cs (hunk 29 above)
Comment (brthor) [inline build_projects/shared-build-targets-utils/BuildPackageVersions.cs]: done.

File: build_projects/dotnet-host-build/PrepareTargets.cs (hunk 25 above)
Comment (brthor) [inline build_projects/dotnet-host-build/PrepareTargets.cs]: done.

File: build_projects/dotnet-cli-build/project.json (hunk 24 above)
Comment (brthor) [inline build_projects/dotnet-cli-build/project.json]: fixed.

File: build_projects/dotnet-cli-build/DebTargets.cs (hunk 23 above)
Comment (brthor) [inline build_projects/dotnet-cli-build/DebTargets.cs]: fixed. There was more cleanup to do as well, thanks for pointing this out.

File: build_projects/dotnet-cli-build/CompileTargets.cs (hunk 22 above)
Comment (brthor) [inline build_projects/dotnet-cli-build/CompileTargets.cs]: made it an option, done.

**Comment by brthor** [comment]
> Should this be removed? It is in the host build.

removed.

**Comment by brthor** [comment]
> Do we need these "SharedFx" variables?

Yes the sdk pkg needs to include the sharedfx component pkg inside of it.

**Comment by brthor** [comment]
PR Feedback taken into account here.

File: build_projects/dotnet-host-build/CompileTargets.cs (hunk 21 above)
Comment (schellap) [inline build_projects/dotnet-host-build/CompileTargets.cs]: We should produce host packages from the feed. Not from the repo at all.

Comment (schellap) [inline build_projects/dotnet-host-build/CompileTargets.cs]: *consume

Comment (brthor) [inline build_projects/dotnet-host-build/CompileTargets.cs]: It will be this way once we move to another repo, this is happening in stages.

File: scripts/update-dependencies/UpdateFilesTargets.cs [hunk 30]
```diff
@@ -190,7 +190,7 @@ private static IEnumerable<JProperty> FindAllDependencyProperties(JObject projec
         [Target]
         public static BuildTargetResult ReplaceCrossGen(BuildTargetContext c)
         {
-            ReplaceFileContents(@"build_projects\dotnet-cli-build\CompileTargets.cs", compileTargetsContent =>
+            ReplaceFileContents(@"build_projects\shared-build-targets-utils\BuildPackageVersions.cs", compileTargetsContent =>
```
Comment (eerhardt) [inline scripts/update-dependencies/UpdateFilesTargets.cs]: This file got renamed.

**Comment by eerhardt** [comment]
:shipit: after fixing update-dependencies for the new path.

**Comment by eerhardt** [comment]
Or fix it separately will work too. Your call

File: scripts/update-dependencies/UpdateFilesTargets.cs (hunk 30 above)
Comment (brthor) [inline scripts/update-dependencies/UpdateFilesTargets.cs]: Will fix, thanks

**Comment by brthor** [comment]
closed in favor of https://github.com/dotnet/cli/pull/3230/files
